src/codegen/entry.cpp

```diff
@@ -118,56 +120,114 @@ static llvm::Module* loadStdlib() {
     return m;
 }
 
-class MyObjectCache : public llvm::ObjectCache {
+class PystonObjectCache : public llvm::ObjectCache {
 private:
-    bool loaded;
+    class HashOStream : public llvm::raw_ostream {
+        unsigned int hash = 0;
+        void write_impl(const char* ptr, size_t size) override { hash = crc32(hash, (const unsigned char*)ptr, size); }
```

Comment:
I don't think a 32-bit hash is good enough :(  That only gives us 64k simultaneous cached files before we expect to get a collision.  And while I doubt we need a cryptographically secure hash here, it seems like it's not going to be performance critical so maybe we should use one just to be safe?  Maybe we can just try SHA-256 and see if it hurts performance at all.